test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -49,6 +50,38 @@ def world_size(self):
     def process_group(self):
         return dist.distributed_c10d._get_default_group()
 
+    @parametrize("use_second_layer", [True, False])
+    def test_fsdp_module_no_compute_grad(self, use_second_layer):
```

Comment:
Can we add a comment about the point of `use_second_layer`?

In my understanding:
- If `use_second_layer == False`, then `b` is not involved in the forward computation.
- If `use_second_layer == True`, then `b` is involved in the forward computation but not connected to the loss that initiates the backward.

Before this PR, would both cases have failed?

Reply:
Will clarify, but yes. Both cases would've failed